dev-docs/security-overview.md

```diff
@@ -0,0 +1,74 @@
+* The [Helm charts used for the installation of the three Constellation services](https://github.com/edgelesssys/constellation/tree/main/internal/constellation/helm/charts/edgeless/constellation-services), which include hashes of the respective containers. *Note: The Helm charts and the hashes are [generated at build time](https://github.com/edgelesssys/constellation/blob/main/internal/constellation/helm/imageversion/imageversion.go). A future version of the CLI will provide a command to print the Helm charts.*
+
+## Cluster creation
+
+When a cluster is [created](https://docs.edgeless.systems/constellation/workflows/create), the CLI interacts with the API of the respective infrastructure provider (e.g., Azure) and launches CVMs with the applicable node image. These CVMs are called *nodes*. On each node, the Bootstrapper is launched.
+
+The CLI automatically selects one of the nodes as *first node*. The CLI automatically verifies the first node's runtime measurements using remote attestation. Based on this, the CLI and the Bootstrapper running on the first node set up a temporary TLS connection. This [aTLS](https://docs.edgeless.systems/constellation/architecture/attestation#attested-tls-atls) connection is mainly used for three things (see the the [interface definition](https://github.com/edgelesssys/constellation/blob/main/bootstrapper/initproto/init.proto) for a comprehensive list of exchanged data):
+1. The CLI sends the hashes of the to-be-installed Kubernetes binaries to the first node.
+2. The CLI generates the  [master secret](../architecture/keys.md#master-secret) of the to-be-created cluster and sends it to the first node.
+3. The first node generates a [kubeconfig file](https://www.redhat.com/sysadmin/kubeconfig) and sends it to the CLI. The kubeconfig file contains Kubernetes credentials for the CLI and the Kubernetes cluster's public key, among others.
+
+After this, the aTLS connection is closed and the node is marked as "initialized". This marker is irrevocably reflected in the node's remote attestation. This mechanism prevents a node from joining different clusters. 
+
+## Kubernetes bootstrapping on the first node
+
+The Bootstrapper on the first node downloads and verifies the Kubernetes binaries, using the hashes it received from the CLI. These binaries include for example [kubelet](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/), [kube-apiserver](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/), and [etcd](https://kubernetes.io/docs/tasks/administer-cluster/configure-upgrade-etcd/). With these, the Bootstrapper creates a single-node Kubernetes cluster. Etcd is a distributed key-value store that Kubernetes uses to store configuration data for services. The etcd agent runs on each control-plane node of a cluster. The agents use mTLS for communication between them. Etcd uses the Raft protocol (over mTLS) to distribute state between nodes.

```

Comment:
Neither the kube-api server nor the etcd is something that we regard as a Kubernetes component: https://github.com/edgelesssys/constellation/blob/main/internal/versions/versions.go#L269

Reply:
These are now also part of the pinned components:

- https://github.com/edgelesssys/constellation/pull/2719
- https://github.com/edgelesssys/constellation/pull/2764

Reply:
Ok sure. I would've not considered them "components that the bootstrapper downloads" since there's an indirection of who downloads the kube-api server, but sure fine with me.